src/gcm_request.erl

```diff
@@ -142,6 +144,16 @@ filter(V) when is_atom(V) ->
 filter(V) when is_integer(V) ->
   integer_to_binary(V).
 
+map_to_binary(Map) when is_map(Map) ->
```

Comment:
The function name (map_to_binary) can not show what it does. The result is still a map instead of a binary.
Maybe atom_to_binary_in_map?